wrw_game/wrw_modules/models.py

```diff
@@ -0,0 +1,178 @@
+        If this value becomes less that 1 (one)
+        the EnemyDown exception is raised.
+        """
+
+        self.health -= 1
+        if self.health < 1:
+            raise exceptions.EnemyDown
+        return self.health
+
+    def select_attack(self):
+        """
+        Return a random attack choice from valid choices.
+        """
+
+        bot_choice = random.choice(settings.CHOICES)
+        return bot_choice
+
+    def select_defence(self):
+        """
+        Return a random defence choice from valid choices.
+        """
+
+        bot_choice = random.choice(settings.CHOICES)
+        return bot_choice
+
+    def __repr__(self) -> str:
+        """
+        Implement a human readable string representation for Enemy class
+        """
+
+        return self.level
+
+    def __str__(self) -> str:
+        """
+        Implement type-casting for Enemy. Enemy cast to string type equals to its level
+        """
+
+        return self.level
+
+
+class Player:
+    """
+    Player class implementation
+    """
+
+    def __init__(self, name: str):
+        """
+        Initialize an player instance
+        """
+
+        self.name = name
+        self.score = 0
+        self.health = settings.PLAYER_HEALTH
+
+    def decrease_health(self):
+        """
+        Method decreases the health points value by 1 (one).
+        If this value becomes less that 1 (one)
+        the GameOver exception is raised.
+        """
+
+        self.health -= 1
+        if self.health < 1:
+            raise exceptions.GameOver
+        return self.health
+
+    def select_attack(self):
+        """
+        Return a fight choice made by the user. Performs choice validation.
+        """
+
+        while True:
+            print('MAKE AN ATTACK CHOICE FROM (WARRIOR - 1, ROBBER - 2, WIZARD - 3): ')
+            player_move = input()
+            if player_move in ('1', '2', '3'):
+                player_choice = settings.CHOICES[int(player_move) - 1]
+                return player_choice
+
+    def select_defence(self):
+        """
+        Return a fight choice made by the user. Performs choice validation.
+        """
+
+        while True:
+            print('MAKE A DEFENCE CHOICE FROM (WARRIOR - 1, ROBBER - 2, WIZARD - 3): ')
+            player_move = input()
+            if player_move in ('1', '2', '3'):
+                player_choice = settings.CHOICES[int(player_move) - 1]
+                return player_choice
+
+    @staticmethod
+    def fight(player_move, bot_move):
+        """
+        Static method to perform a fight.
+        Takes two arguments representing attack and defence choices.
+        Performs fight result calculation and return it back.
+        """
+
+        if player_move == bot_move:
+            return "draw"
+        battle = (player_move, bot_move)
+        if battle in settings.WIN_CONDITION:
+            return True
+        return False
+
+    def attack(self, enemy, player_move, bot_move):
```

Comment:
А нащо тобі оті moves у якості аргументів?
Виходить, що ти робиш dep-injection.
```suggestion
    def attack(self, enemy):
```

А от атаку/захист можна витягти з потрібного методу обʼєктів:
```python
attack = self.select_attack()
defence = enemy.select_defence()
fight_result = self.fight(attack, defence)
```

І не потрібно десь паритись за те, щоб витягти щось та прокинути до методу.